app/javascript/components/rules/RuleEditorHeader.vue

```diff
@@ -283,6 +283,10 @@ export default {
       type: String,
       default: "",
     },
+    currentUserId: {
```

Comment:
Why didn't the linter catch the undeclared variable use of 'currentUserId' (first instance I'm seeing is old 323/new 327)?  Is it defined with a default value somewhere that gets overridden by the prop?

If it's a problem with the linter, we should make any appropriate edits to the eslint config or to how we call eslint in the scripts section of the package.json so this doesn't happen again.

Reply:
There's a rule I added to the eslint config to spot the undefined prop, "vue/no-undef-properties". The only issue is that it does not detect properties defined in files or mixins. So might give several false positives.
I'am adding it to the commit, so now it will to the developer to go through the errors this rule generates and identify which ones are true positives (real errors).

Reply:
Seems like you put it in and then removed it cause it was providing too many false positives due to our extensive use of mixins, which is fine.  Just means that we now know that we gotta be very careful about this sort of situation in the future.